src/store/reducers.ts

```diff
@@ -291,15 +291,32 @@ export const errorsReducer = handleActions(
 
 
 // FIXME Move all the below (potentially the fetchCount ones too) to their own file
+export interface TaskSettings {
+  showCompleted: boolean;
+  sortOrder: string;
+  filterBy: string;
+  searchTerm: string;
+}
+
 export interface SettingsType {
   locale: string;
+  tasks: TaskSettings;
 }
 
 export const settingsReducer = handleActions(
   {
-    [actions.setSettings.toString()]: (_state: {key: string | null}, action: any) => (
-      { ...action.payload }
+    [actions.setSettings.toString()]: (state: SettingsType, action: any) => (
+      { ...state, ...action.payload }
     ),
   },
-  { locale: 'en-gb' }
+  {
+    locale: 'en-gb',
+
+    tasks: {
+      showCompleted: false,
+      sortOrder: 'dueDate',
```

Comment:
I'm not sure if you took a look at the code I sent you back then, but I assume you have some sort of a "smart" sort, that should be the default, not this.